### PR TITLE
Implementation of builtin_hooks: check_ssh, port_up, ping

### PR DESCRIPTION
**Pull request by samvarankashyap**
Summary:
Added playbooks structure for built_in hooks a.k.a global_hooks
-     check_ssh --> uses ansible ping module to check ssh connection
-     port_up --> uses ansible wait_for module to wait on list of ports availability
-     ping --> uses ansible command module to delegate ICMP ping requests

Example: Pinfile 
```yaml
---
os-server-addl-vols:
  topology:
    topology_name: os-server-inst
    resource_groups:
      - resource_group_name: os-server-addl-vols
        resource_group_type: openstack
        resource_definitions:
          - name: "database"
            role: os_server
            flavor: m1.small
            image: CentOS-7-x86_64-GenericCloud-1612
            count: 1
            keypair: testkeypair_sk
            fip_pool: 10.8.240.0
            networks:
              - e2e-openstack
        credentials:
          filename: clouds.yaml
          profile: ci-rhos
  layout:
    inventory_layout:
      vars:
        hostname: __IP__
      hosts:
        addl-vols-node:
          count: 1
          host_groups:
            - hello
  hooks:
    postup:
      # check_ssh, ping and port_up are builtin hooks
      # note builtin hooks follow different structure when compared to localhooks
      - name: check_ssh
        extra_vars:
          # since checking ssh depends on logging into machine pem file, ssh_user are must
          ansible_ssh_private_key_file:  /home/srallaba/workspace/lp_ws/ex_hooks/os_workspace/keys/testkeypair_sk.pem
          ansible_ssh_user: centos
      - name: ping
      - name: port_up
        extra_vars:
          ports:
            - 22
      - name: osoos
        type: ansible
        context: True
        actions:
          - playbook: install_packages.yaml
            extra_vars:
              ansible_ssh_private_key_file: "/home/srallaba/workspace/lp_ws/ex_hooks/os_workspace/keys/testkeypair_sk.pem"
              ansible_ssh_user: centos
          - playbook: git_clone.yaml
            extra_vars:
              ansible_ssh_private_key_file: "/home/srallaba/workspace/lp_ws/ex_hooks/os_workspace/keys/testkeypair_sk.pem"
              ansible_ssh_user: centos
```
Example workspace link: https://github.com/samvarankashyap/linchpin_hooks_openstack_ws/tree/built_in_hooks

fixes: #174 

to_be_reviewed and merged after #689 is merged

### CODE REVIEW

**Comment by 14rcole** [comment]
@samvarankashyap is there any reason why we haven't merged this yet?

**Comment by samvarankashyap** [comment]
@14rcole : Need to be re-based with respect to recent changes .

**Comment by 14rcole** [comment]
@samvarankashyap If you can do that, I'll give it a review.  It's been in the pipeline for a while now

**Comment by 14rcole** [comment]
@samvarankashyap If you can do that, I'll give it a review.  It's been in the pipeline for a while now

**Comment by samvarankashyap** [comment]
@14rcole Rebased !

File: linchpin/hooks/built_ins/check_ssh/check_ssh.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+- hosts: all
+  gather_facts: false
+  tasks:
+    - name: ping task
```
Comment (p3ck) [inline linchpin/hooks/built_ins/check_ssh/check_ssh.yaml]: How is this checking ssh?

Comment (samvarankashyap) [reply linchpin/hooks/built_ins/check_ssh/check_ssh.yaml]: @p3ck : Ansible ping module does a ssh connection to the host. https://docs.ansible.com/ansible/latest/modules/ping_module.html

**Comment by samvarankashyap** [comment]
"msg": "SELinux policy denies access."
}
	to retry, use: --limit @/usr/lib/python2.7/site-packages/linchpin/provision/libvirt.retry

PLAY RECAP *********************************************************************
localhost                  : ok=47   changed=14   unreachable=0    failed=1   

https://jenkins-continuous-infra.apps.ci.centos.org/job/ci-linchpin/1293/artifact/linchpin/centos7_logs/libvirt.log

**Comment by samvarankashyap** [comment]
[test]

**Comment by samvarankashyap** [comment]
@p3ck @14rcole Did SELinux policy change recently w.r.to containers? ^^^^

**Comment by 14rcole** [comment]
@samvarankashyap no idea

**Comment by samvarankashyap** [comment]
[test]

**Comment by samvarankashyap** [comment]
@14rcole @p3ck needs re-review.

File: config/Dockerfiles/tests.d/libvirt/05_libvirt-builtins [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+echo "Test Name: ${TEST_NAME}"
+echo "${DESCRIPTION}"
+
+TARGET="libvirt-builtins"
+
+pushd docs/source/examples/workspaces/libvirt-global-hooks
+
+echo "Testing check the builtins port_up"
+
+find / -name "*port_up.yaml"
```
Comment (p3ck) [inline config/Dockerfiles/tests.d/libvirt/05_libvirt-builtins]: what is the purpose of the find?  Is this left over debug?

**Comment by 14rcole** [comment]
@samvarankashyap I noticed that you added some libvirt hooks in the workspaces.  Are all of the built-in hooks provider-agnostic?

**Comment by samvarankashyap** [comment]
@14rcole All built-in hooks are provider-agnostic. I added test inside libvirt because it is easier to test and will not eat up external provider resources for testing.

File: config/Dockerfiles/tests.d/libvirt/05_libvirt-builtins (hunk 2 above)
Comment (samvarankashyap) [reply config/Dockerfiles/tests.d/libvirt/05_libvirt-builtins]: Its a left over from debug. I will remove it :+1: good catch.

**Comment by samvarankashyap** [comment]
[merge]

**Pull request merged**